src/audio/host.c

```diff
@@ -285,13 +285,17 @@ static int host_trigger(struct comp_dev *dev, int cmd)
 	case COMP_TRIGGER_STOP:
 		ret = host_stop(dev);
 		/* host stop fails, let's emit the error log */
-		if (ret)
-			trace_host_error("host: stop failed %d", ret);
+		if (ret < 0)
+			trace_host_error("host_trigger(): host stop failed: %d",
+					 ret);
```

Comment:
So if the host_stop is failed.
We will fall through to stop DMA?

Reply:
Yes, based on Liam's requirement.

Reply:
@keyonjie 
Yes, you are right. This path is for: in case we will modify the host_stop function, we should correctly handle the error.

Actually, when we are stopping, even host stops failed, we should also stop DMA. They are independently, right? So I agree with Liam to fall through to stop DMA.